src/ch/ehi/sqlgen/generator_impl/jdbc/GeneratorGeoPackage.java

```diff
@@ -331,15 +333,131 @@ public void visit1TableEnd(DbTable tab) throws IOException {
+					String triggerStmt3 = "CREATE TRIGGER \"rtree_" + tab.getName().getName() + "_" + geo.getName() + "_update2\" AFTER UPDATE OF \"" + geo.getName() + "\" ON \"" + tab.getName().getName() + "\" "
+					+ "WHEN OLD.\"" + primaryKey + "\" = NEW.\"" + primaryKey + "\" AND (NEW.\"" + geo.getName() + "\" ISNULL OR ST_IsEmpty(NEW.\"" + geo.getName() + "\")) "
+					+ "BEGIN "
+					+ "DELETE FROM \"rtree_" + tab.getName().getName() + "_" + geo.getName() + "\" WHERE id = OLD.\"" + primaryKey + "\"; "
+					+ "END";
+
+					addCreateLine(new Stmt(triggerStmt3));
+
+					/* Conditions: Update of any column
+					 * Row ID change
+					 * Non-empty geometry
+					 * Actions: Remove record from rtree for old <i>
+					 * Insert record into rtree for new <i> */
+					String triggerStmt4 = "CREATE TRIGGER \"rtree_" + tab.getName().getName() + "_" + geo.getName() + "_update3\" AFTER UPDATE ON \"" + tab.getName().getName() + "\" "
+					+ "WHEN OLD.\"" + primaryKey + "\" != NEW.\"" + primaryKey + "\" AND (NEW.\"" + geo.getName() + "\" NOTNULL AND NOT ST_IsEmpty(NEW.\"" + geo.getName() + "\")) "
+					+ "BEGIN "
+					+ "DELETE FROM \"rtree_" + tab.getName().getName() + "_" + geo.getName() + "\" WHERE id = OLD.\"" + primaryKey + "\"; "
+					+ "INSERT OR REPLACE INTO \"rtree_" + tab.getName().getName() + "_" + geo.getName() + "\" VALUES ("
+					+ "NEW.\"" + primaryKey + "\",ST_MinX(NEW.\"" + geo.getName() + "\"),ST_MaxX(NEW.\"" + geo.getName() + "\"),ST_MinY(NEW.\"" + geo.getName() + "\"),ST_MaxY(NEW.\"" + geo.getName() + "\")); "
+					+ "END";
+
+					addCreateLine(new Stmt(triggerStmt4));
+
+					/* Conditions: Update of any column
+					 * Row ID change
+					 * Empty geometry
+					 * Actions: Remove record from rtree for old and new <i> */
+					String triggerStmt5 = "CREATE TRIGGER \"rtree_" + tab.getName().getName() + "_" + geo.getName() + "_update4\" AFTER UPDATE ON \"" + tab.getName().getName() + "\" "
+					+ "WHEN OLD.\"" + primaryKey + "\" != NEW.\"" + primaryKey + "\" AND (NEW.\"" + geo.getName() + "\" ISNULL OR ST_IsEmpty(NEW.\"" + geo.getName() + "\")) "
+					+ "BEGIN "
+					+ "DELETE FROM \"rtree_" + tab.getName().getName() + "_" + geo.getName() + "\" WHERE id IN (OLD.\"" + primaryKey + "\",NEW.\"" + primaryKey + "\"); "
+					+ "END";
+
+					addCreateLine(new Stmt(triggerStmt5));
+
+					/* Conditions: Row deleted
+					 * Actions: Remove record from rtree for old <i>
+					 */
+					String triggerStmt6 = "CREATE TRIGGER \"rtree_" + tab.getName().getName() + "_" + geo.getName() + "_delete\" AFTER DELETE ON \"" + tab.getName().getName() + "\" "
+					+ "WHEN OLD.\"" + geo.getName() + "\" NOT NULL BEGIN "
+					+ "DELETE FROM \"rtree_" + tab.getName().getName() + "_" + geo.getName() + "\" WHERE id = OLD.\"" + primaryKey + "\"; "
+					+ "END";
+
+					addCreateLine(new Stmt(triggerStmt6));
+
+					String[] triggerStmts = {triggerStmt1, triggerStmt2, triggerStmt3, triggerStmt4, triggerStmt5, triggerStmt6};
```

Comment:
Fehlen noch die DROP TRIGGER statements

Reply:
Die DROP TRIGGER statements sollten jetzt vorhanden sein. Ich denke, dass in SQLite Trigger automatisch löscht, wenn die zugehörige Tabelle gelöscht wird ("Note that triggers are automatically dropped when the associated table is dropped." ), deshalb vorsichtshalber mit IF EXISTS.